v2/googlecloud-to-googlecloud/src/main/java/com/google/cloud/teleport/v2/templates/bigtablechangestreamstopubsub/schemautils/PubSubUtils.java

```diff
@@ -0,0 +1,407 @@
+    changelogEntryMessageProto =
+        ChangelogEntryMessageProto.ChangelogEntryProto.newBuilder()
+            .setRowKey(
+                ByteString.copyFrom(
+                    (byte[])
+                        FORMATTERS.get(PubSubFields.ROW_KEY_BYTES).format(this, changeJsonParsed)))
+            .setModType(
+                ChangelogEntryMessageProto.ChangelogEntryProto.ModType.valueOf(
+                    (String) FORMATTERS.get(PubSubFields.MOD_TYPE).format(this, changeJsonParsed)))
+            .setIsGC((Boolean) FORMATTERS.get(PubSubFields.IS_GC).format(this, changeJsonParsed))
+            .setTieBreaker(
+                (Integer) FORMATTERS.get(PubSubFields.TIEBREAKER).format(this, changeJsonParsed))
+            .setColumnFamily(
+                (String) FORMATTERS.get(PubSubFields.COLUMN_FAMILY).format(this, changeJsonParsed))
+            .setCommitTimestamp(
+                (Long) FORMATTERS.get(PubSubFields.COMMIT_TIMESTAMP).format(this, changeJsonParsed))
+            .setColumn(
+                ByteString.copyFrom(
+                    (byte[])
+                        FORMATTERS.get(PubSubFields.COLUMN_BYTES).format(this, changeJsonParsed)))
+            .setTimestamp(
+                (Long) FORMATTERS.get(PubSubFields.TIMESTAMP).format(this, changeJsonParsed))
+            .setTimestampFrom(
+                (Long) FORMATTERS.get(PubSubFields.TIMESTAMP_FROM).format(this, changeJsonParsed))
+            .setTimestampTo(
+                (Long) FORMATTERS.get(PubSubFields.TIMESTAMP_TO).format(this, changeJsonParsed))
+            .setValue(
+                ByteString.copyFrom(
+                    (byte[])
+                        FORMATTERS.get(PubSubFields.VALUES_BYTES).format(this, changeJsonParsed)))
+            .setSourceInstance(
+                (String)
+                    FORMATTERS.get(PubSubFields.SOURCE_INSTANCE).format(this, changeJsonParsed))
+            .setSourceCluster(
+                (String) FORMATTERS.get(PubSubFields.SOURCE_CLUSTER).format(this, changeJsonParsed))
+            .setSourceTable(
+                (String) FORMATTERS.get(PubSubFields.SOURCE_TABLE).format(this, changeJsonParsed))
+            .build();
+
+    Publisher publisher = null;
+    PubsubMessage.Builder message = PubsubMessage.newBuilder();
+
+    switch (messageEncoding) {
+      case "BINARY":
+        message.setData(changelogEntryMessageProto.toByteString());
+        break;
+      case "JSON":
+        String jsonString =
+            JsonFormat.printer()
+                .omittingInsignificantWhitespace()
+                .print(changelogEntryMessageProto);
+        message.setData(ByteString.copyFromUtf8(jsonString));
+        break;
+      default:
+        final String errorMessage =
+            "Invalid message encoding: "
+                + messageEncoding
+                + ". Supported output formats: JSON, BINARY";
+        throw new IllegalArgumentException(errorMessage);
```

Comment:
This becomes a bit repetitive. Can we please parse MessageEncoding into an enum, validate it there and have two cases with default throwing something like:

`thow new IllegalStateException("Unknown message encoding: " + messageEncoding);`